### DISCUSSION TITLE
How to replace the default traceId in skywalking.The system already has traceId.

### DISCUSSION THREAD

**Discussion by etertal** [apache/skywalking#12005] [General]
![后端微服务架构图](https://github.com/apache/skywalking/assets/16147158/1a841815-600b-4037-863c-38e5b2dfa48e)

Our current system generates traceId and spanId through Spring-cloud-sleuth. In the case of mobile devices, traceId and spanId are generated by the mobile device. The entire link and log are concatenated by sleuth's traceId. Now we want to introduce skywalking to visualize the entire request link and replace the default traceId-segmentId generated by skywalking with the original traceId-spanId,  I'm not sure if there's a problem with what I described. 

Based on the scenario I described above, I would like to use Skywalking, but I need to keep the original traceId and spanId. What do I need to do?

**Comment by wu-sheng** [comment]
First of all, segment could NOT be replaced. This will not be a driop-in, for several reasons.
1. Trace ID, segment ID and numberic span ID are required by the protocol and all agent kernels.
2. Those are all required with endpoint name and service/instance names of downstream in the header are key parts to run analysis. You could read STAM for more details.

From the graph of arch you shared, I want to point out these
- SkyWalking never supports traces in logs, only trace ID when backend OAP is up and running. Only Java agent has a mode to do so.
- ELK is not our scope either.

The key question you are asking does not make sense, all of your infra is working, why do you need SkyWalking to replace existing things? The gap is clear, we are seeing distributed tracing in different ways.

**Reply by etertal** [reply]
Thank you very much for your reply. My thinking was wrong. I struggled for a while and finally understood.
